src/components/InputCalendar/index.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from 'react'
+import {
+  getSeconds,
+  setSeconds,
+  getMinutes,
+  setMinutes,
+  getHours,
+  setHours,
+  isEqual,
+  isDate,
+} from 'date-fns'
+import { noop, isNull, isUndefined, isFunction } from 'lodash'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+
+import chain from 'helpers/generators/chain'
+import Label from '../Typographies/Label'
+import Input from '../Input'
+
+import StyledTimeContainer from './styles/StyledTimeContainer'
+
+/**
+ * A select field.
+ *
+ */
+const InputTime = ({
+  className,
+
+  hoursLabel,
+  minutesLabel,
+  secondsLabel,
+  datePartRender,
+
+  defaultValue,
+  value,
+  onChange,
+}) => {
+  const [stateValue, setStateValue] = useState(defaultValue)
+
+  // Sync with value if in controlled state
+  if (isDate(value) && !isEqual(stateValue, value)) {
+    setStateValue(value)
```

Comment:
I'm pretty sure you can use `useEffect` here. Because now you're blocking UI thread for the rendering